src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java

```diff
@@ -52,11 +56,13 @@
 
 
   public AsyncReportStatusListener( final String path,
-                             final UUID uuid,
-                             final String mimeType ) {
+                                    final UUID uuid,
+                                    final String mimeType,
+                                    final List<? extends ReportProgressListener> callbackListeners ) {
     this.path = path;
     this.uuid = uuid;
     this.mimeType = mimeType;
+    this.callbackListeners = callbackListeners;
 
```

Comment:
Always check your list parameters for null. Dont allow it, demand that they pass in an empty list instead. Then you dont have to check for null all the time.

Do you expect that users are allowed to modify that list after it has been set? If not, make a copy for your private use to isolate yourself from evil user actions.